From 930026ba571740c0a929eacdf6bf540f19359caf Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 4 Jul 2019 22:12:11 -0700
Subject: [PATCH 0642/1851] v4l: xilinx: sdirxss: Check for 10bpc in dt

commit c871383af5b02e6c3e6d331dadc64d1b3452d266 from
https://github.com/Xilinx/linux-xlnx.git

This patch adds check for xlnx,video-width to be 10 as the SDI Rx
Subsystem supports only 10 bpc.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-sdirxss.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-sdirxss.c b/drivers/media/platform/xilinx/xilinx-sdirxss.c
index d6f0b1da6482..07dfaa94001a 100644
--- a/drivers/media/platform/xilinx/xilinx-sdirxss.c
+++ b/drivers/media/platform/xilinx/xilinx-sdirxss.c
@@ -1805,7 +1805,7 @@ static int xsdirxss_parse_of(struct xsdirxss_state *xsdirxss)
 			format->vf_code, format->width, format->bpp);
 
 		if (format->vf_code != XVIP_VF_YUV_422 &&
-		    format->vf_code != XVIP_VF_YUV_420) {
+		    format->vf_code != XVIP_VF_YUV_420 && format->width != 10) {
 			dev_err(core->dev,
 				"Incorrect UG934 video format set.\n");
 			return -EINVAL;
-- 
2.31.1

